### PR TITLE
Correct bcrypt salt regex

### PR DESCRIPTION
**Pull request by sabo**
This corrects the regex used by the pw_hash function for bcrypt-type hashes. The first two characters are interpreted as a strength parameter by bcrypt, and must take a value between 4 and 31 inclusive.

### CODE REVIEW

**Comment by jordanbreen28** [comment]
Hi @sabo - thanks for submitting this PR. Whilst the changes look good, could I ask that you look to resolve the present test failures and sign our CLA so we can proceed with this. :-) 
Thanks!

**Comment by jordanbreen28** [comment]
@sabo - please disregard my previous comment regarding the test failures, the failures present at first appeared to be transient. (With exception of Windows 2012 R2, these tests are regularly failing in our nightlies and not thought to be due to your PR)

Could you please rebase with the current main branch, and sign our CLA, so that we can proceed with this.
Thanks!

**Comment by sabo** [comment]
Hi @jordanbreen28, working on getting approval for that CLA.

**Comment by jordanbreen28** [comment]
Great @sabo - in the meantime could you rebase this PR with the current main :-)

**Comment by sabo** [comment]
@jordanbreen28, branch rebased and CLA signed, should be good to go

**Comment by jordanbreen28** [comment]
@sabo Great work :-) Just one more small thing before merging, could you please squash your commits into a single commit for this PR.

**Comment by kenyon** [comment]
You can squash on merge using GitHub's web interface.

**Review by jordanbreen28 — APPROVED** [review]

**Pull request merged**
